### PR TITLE
viewmail.php token for quarantine reports

### PR DESCRIPTION
**Pull request by shawniverson**
Fix for Issue #664

### CODE REVIEW

File: MailScanner_perl_scripts/MailWatch.pm [hunk 1]
```diff
@@ -345,7 +347,13 @@ sub MailWatchLogging {
     my ($todomain, @todomain);
     @todomain = @{$message->{todomain}};
     $todomain = $todomain[0];
-
+    
+    # Generate token for mail viewing
+    my ($token, $sha1);
+    $sha1 = Digest::SHA1->new;
+    $sha1->add($message->{id}, $timestamp, $message->{size});
```
Comment (endelwar) [inline MailScanner_perl_scripts/MailWatch.pm]: what about adding `headers` to the hash for more uniqueness?

Comment (shawniverson) [inline MailScanner_perl_scripts/MailWatch.pm]: Good idea, adding.

File: create.sql [hunk 2]
```diff
@@ -125,6 +125,7 @@ CREATE TABLE IF NOT EXISTS `maillog` (
   `headers` mediumtext COLLATE utf8_unicode_ci,
   `quarantined` tinyint(1) DEFAULT '0',
   `rblspamreport` mediumtext COLLATE utf8_unicode_ci DEFAULT NULL,
+  `token` mediumtext COLLATE utf8_unicode_ci DEFAULT NULL,
```
Comment (endelwar) [inline create.sql]: `CHAR(40)` (or `CHAR(64)` for future sha256 support) should be better than mediumtext

Comment (shawniverson) [inline create.sql]: yeah, CHAR(64) sounds like a good idea

**Pull request merged**